cli/command/service/opts.go

```diff
@@ -469,33 +499,83 @@ func (opts *serviceOptions) ToService() (swarm.ServiceSpec, error) {
 					Search:      opts.dnsSearch.GetAll(),
 					Options:     opts.dnsOption.GetAll(),
 				},
-				Hosts:           convertExtraHostsToSwarmHosts(opts.hosts.GetAll()),
-				StopGracePeriod: opts.stopGrace.Value(),
-				Secrets:         nil,
-				Healthcheck:     healthConfig,
+				Hosts:       convertExtraHostsToSwarmHosts(opts.hosts.GetAll()),
+				Secrets:     nil,
+				Healthcheck: healthConfig,
 			},
-			Networks:      convertNetworks(opts.networks.GetAll()),
-			Resources:     opts.resources.ToResourceRequirements(),
-			RestartPolicy: opts.restartPolicy.ToRestartPolicy(),
+			Networks:  convertNetworks(opts.networks.GetAll()),
+			Resources: opts.resources.ToResourceRequirements(),
 			Placement: &swarm.Placement{
 				Constraints: opts.constraints.GetAll(),
 				Preferences: opts.placementPrefs.prefs,
 			},
 			LogDriver: opts.logDriver.toLogDriver(),
 		},
-		Networks:       convertNetworks(opts.networks.GetAll()),
-		Mode:           serviceMode,
-		UpdateConfig:   opts.update.config(),
-		RollbackConfig: opts.rollback.config(),
-		EndpointSpec:   opts.endpoint.ToEndpointSpec(),
+		Networks:     convertNetworks(opts.networks.GetAll()),
+		Mode:         serviceMode,
+		EndpointSpec: opts.endpoint.ToEndpointSpec(),
 	}
 
+	if flags.Changed(flagStopGracePeriod) {
+		service.TaskTemplate.ContainerSpec.StopGracePeriod = opts.stopGrace.Value()
+	}
+	if anyChanged(flags, flagRestartDelay, flagRestartMaxAttempts, flagRestartWindow, flagRestartCondition) {
+		service.TaskTemplate.RestartPolicy = opts.restartPolicy.defaults()
```

Comment:
I would love to find a way to avoid needing `flags` here, but I can't find any right now.

Can we move this these back to the `ToRestartPolicy()` function, and add a `flags` arg ? (and the same with the few below this)